src/test/java/org/jenkinsci/plugins/workflow/test/steps/SemaphoreStep.java

```diff
@@ -222,7 +228,7 @@ public static class Execution extends AbstractStepExecutionImpl {
         @Override public void stop(Throwable cause) throws Exception {
             State s = State.get();
             synchronized (s) {
-                s.contexts.remove(k);
+                s.keyStates.remove(k);
```

Comment:
I am not sure that this behavior is the same. Would have to think about it some more.

Reply:
Seems safer to switch to the failure state. Should be idempotent right? There is no real reason to remove entries from the map.

Reply:
Adjusted in [d05eafd](https://github.com/jenkinsci/workflow-support-plugin/pull/260/commits/d05eafd27820d48db69890d04658053da4887743)